### PR TITLE
Merge the clear-sky ABI gsrcsr python version bufr2ioda python file a…

### PR DESCRIPTION
**Pull request by xyzemc**
This PR is addressing issure #221  to convert the clear sky GOES ABI radiance bufr data to ioda. 
Two files have been added to RDASApp/rrfs_test/IODA/python.

### CODE REVIEW

**Comment by SamuelDegelia-NOAA** [comment]
@xyzemc Thanks for adding the new converter. Would you like to stage the ABI IODA files in `RDAS_DATA`?

**Comment by xyzemc** [comment]
> @xyzemc Thanks for adding the new converter. Would you like to stage the ABI IODA files in `RDAS_DATA`?

Yes. Thanks for reminding me.
Please copy the following data from the directory: /scratch2/NCEPDEV/fv3-cam/Xiaoyan.Zhang/noscrub/JEDI/RDASApp_1107/expr/mpas_2024052700/data/obs
```
abi_g16.satbias.nc4      abi_g16.tlapse.txt   abi_g18.satbias_cov.nc4  ioda_abi_g16.nc@
abi_g16.satbias_cov.nc4  abi_g18.satbias.nc4  abi_g18.tlapse.txt       ioda_abi_g18.nc@
```

**Comment by SamuelDegelia-NOAA** [comment]
> > @xyzemc Thanks for adding the new converter. Would you like to stage the ABI IODA files in `RDAS_DATA`?
> 
> Yes. Thanks for reminding me. Please copy the following data from the directory: /scratch2/NCEPDEV/fv3-cam/Xiaoyan.Zhang/noscrub/JEDI/RDASApp_1107/expr/mpas_2024052700/data/obs
> 
> ```
> abi_g16.satbias.nc4      abi_g16.tlapse.txt   abi_g18.satbias_cov.nc4  ioda_abi_g16.nc@
> abi_g16.satbias_cov.nc4  abi_g18.satbias.nc4  abi_g18.tlapse.txt       ioda_abi_g18.nc@
> ```

Will do! Do you plan to include channel 17?

**Comment by xyzemc** [comment]
> > > @xyzemc Thanks for adding the new converter. Would you like to stage the ABI IODA files in `RDAS_DATA`?
> > 
> > 
> > Yes. Thanks for reminding me. Please copy the following data from the directory: /scratch2/NCEPDEV/fv3-cam/Xiaoyan.Zhang/noscrub/JEDI/RDASApp_1107/expr/mpas_2024052700/data/obs
> > ```
> > abi_g16.satbias.nc4      abi_g16.tlapse.txt   abi_g18.satbias_cov.nc4  ioda_abi_g16.nc@
> > abi_g16.satbias_cov.nc4  abi_g18.satbias.nc4  abi_g18.tlapse.txt       ioda_abi_g18.nc@
> > ```
> 
> Will do! Do you plan to include channel 17?

Do you mean GOES_17?

**Comment by SamuelDegelia-NOAA** [comment]
Oops, I see those are satellites now. I first read them and thought the files were split by channel. You can ignore this!

**Comment by SamuelDegelia-NOAA** [comment]
The data have been staged on all supported machines. 

Just out of my curiosity, are the ABI data provided with a parallax correction applied? Or is that something that we can apply through JEDI filters?

**Comment by xyzemc** [comment]
> The data have been staged on all supported machines.
> 
> Just out of my curiosity, are the ABI data provided with a parallax correction applied? Or is that something that we can apply through JEDI filters?

Please remind me what the parallax correction is. The abi data we used in operational is the 16x16 averaged products (CSR), instead of the original pixel data.

**Comment by SamuelDegelia-NOAA** [comment]
The parallax correction adjusts for location errors due to tall clouds. See [here](https://satelliteliaisonblog.com/2017/04/19/why-is-my-goes-16-imagery-displaced-parallax/). Since this PR is for clear-sky ABI it should not be relevant, but once we get to all-sky assimilation then we might need to think about it if the data are not already corrected.

**Comment by xyzemc** [comment]
> The parallax correction adjusts for location errors due to tall clouds. See [here](https://satelliteliaisonblog.com/2017/04/19/why-is-my-goes-16-imagery-displaced-parallax/). Since this PR is for clear-sky ABI it should not be relevant, but once we get to all-sky assimilation then we might need to think about it if the data are not already corrected.

That's a good point. Let's think about it when we implement all-sky in the near future.

File: rrfs-test/IODA/python/bufr2ioda_gsrcsr.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "data_format"      : "bufr_d",
+  "data_type"        : "gsrcsr",
+  "cycle_type"       : "rap",
+  #"cycle_type"       : "{{ RUN }}",
+  "cycle_datetime"   : "{{ current_cycle | to_YMDH }}",
+  "dump_directory"   : "{{ DMPDIR }}",
+  "ioda_directory"   : "{{ COM_OBS }}",
+  "subsets"          : ['NC021046'],
```
Comment (delippi) [inline rrfs-test/IODA/python/bufr2ioda_gsrcsr.json]: Out of curiosity, how/why are these highlighted in red? Is line 5 needed? It looks like it is commented out.

Other than that it looks good to me!

**Review by delippi — DISMISSED** [review]

**Review by hu5970 — DISMISSED** [review]

**Comment by hu5970** [comment]
@xyzemc Could you update this branch so we can merge it. Thanks, Ming

File: rrfs-test/IODA/python/bufr2ioda_gsrcsr.json (hunk 1 above)
Comment (xyzemc) [reply rrfs-test/IODA/python/bufr2ioda_gsrcsr.json]: Thanks. I have deleted the line 5.

**Comment by xyzemc** [comment]
> @xyzemc Could you update this branch so we can merge it. Thanks, Ming

I have updated this branch.

**Review by delippi — APPROVED** [review]

**Review by PraveenKumar-NOAA — APPROVED** [review]
Looks good!

**Review by hu5970 — APPROVED** [review]

**Pull request merged**